packages/react-storage/src/components/StorageBrowser/views/Controls/Search.tsx

```diff
@@ -1,33 +1,62 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 
 import { Field } from '../../components/Field';
-import {
-  ButtonElement,
-  IconElement,
-  InputElement,
-  LabelElement,
-  SpanElement,
-} from '../../context/elements';
+import { ButtonElement, IconElement } from '../../context/elements';
 
 import { CLASS_BASE } from '../constants';
 
 const BLOCK_NAME = `${CLASS_BASE}__search`;
 
 const TOGGLE_BLOCK = 'toggle';
 
-export const SearchControl = (): React.JSX.Element => (
-  <Field
-    icon={<IconElement variant="search" />}
-    className={BLOCK_NAME}
-    variant="search"
-  >
-    <ButtonElement className={`${BLOCK_NAME}__button`} />
-    <SpanElement className={`${BLOCK_NAME}-${TOGGLE_BLOCK}__container`}>
-      <InputElement
-        className={`${BLOCK_NAME}-${TOGGLE_BLOCK}__checkbox`}
-        type="checkbox"
-      />
-      <LabelElement className={`${BLOCK_NAME}-${TOGGLE_BLOCK}__label`} />
-    </SpanElement>
-  </Field>
-);
+interface SearchControlProps {
+  onSearch: (query: string) => void;
+  placeholder?: string;
+}
+
+export const SearchControl = ({
+  placeholder,
+  onSearch,
+}: SearchControlProps): React.JSX.Element => {
+  const [inputValue, setInputValue] = useState('');
```

Comment:
i plan to make this component uncontrolled, but i didn't want to deal with ref forwarding just yet.